File: 0006-zigzag-conversion/0006-zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if (s.length()<=numRows || numRows==1) {
            return s;
        }

        String ans="";
        boolean frontDir=true;
        int row=0;
        ArrayList<String> trial=new ArrayList<>();
        for (int i=0; i<numRows; i++) {
            trial.add(new String());
        }

        for (int i=0; i<s.length(); i++) {
            String tmp=trial.get(row);
            tmp+=s.charAt(i);
            trial.set(row, tmp);
            if (frontDir) {
                row++;
            }
            else {
                row--;
            }

            if (row==numRows-1) {
                frontDir=false;
            }
            else if (row==0) {
                frontDir=true;
            }
        }

        for (int i=0; i<numRows; i++) {
            ans+=trial.get(i);
        }

        return ans;
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if len(s)<=numRows or numRows==1:
            return s
        ans=""
        container=[[''] for i in range(numRows)]
        for i in range(len(s)):
            tmp=i%(numRows*2-2)
            if tmp>numRows-1:
                container[numRows*2-tmp-2].append(s[i])
            else:
                container[tmp].append(s[i])

        for i in container:
            for j in i[1::]:
                ans+=j
        return ans